app.rb

```diff
@@ -0,0 +1,247 @@
+  end
+
+  def can_use_services?
+    true
+  end
+end
+
+class Main
+  def initialize
+    @book = []
+    @people = []
+    @rental = []
+  end
+
+  def menu
+    puts "1 - List all books\n2 - List all people\n3 - Create a person\n4 - Create a book
+            \n5 - Create a rental\n6 - List all rental for a given id\n7 - Exit"
+    input
+  end
+
+  def input
+    action = gets.chomp.to_i
+    if action < 7
+      select(action)
+    elsif action == 7
+      puts 'Bye!'
+    else
+      puts 'Please select a number from 1 to 7'
+      input
+    end
+  end
+
+  def select(action)
+    case action
+    when 1
+      list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_a_person
+    when 4
+      create_a_book
+    when 5
+      create_a_rental
+    when 6
+      list_all_rentals
+    end
+  end
+
+  def list_all_books
+    if @book.empty?
+      (puts "\n\nThere are no books\n\n")
+    else
+      @book.each { |b| puts "\n\n#{b.title} written by #{b.author}\n\n" }
+    end
+    menu
+  end
+
+  def list_all_people
+    if @people.empty?
+      (puts "\n\nThere are no people\n\n")
+    else
+      @people.each { |p| puts "\n\nName:#{p.name}\nAge:#{p.age}\nClass:#{p.class}\nID:#{p.id}\n\n" }
+    end
+    menu
+  end
+
+  def create_a_person
+    puts "\n\nCreate a Student (1) or create a Teacher (2)\n\n"
+    x = gets.chomp.to_i
+    case x
+    when 1
+      create_a_student
+    when 2
+      create_a_teacher
+    else
+      create_a_person
+    end
+  end
+
+  def create_a_student
+    puts 'classroom'
+    cls = gets.chomp
+    puts 'age'
+    age = gets.chomp.to_i
+    puts 'name'
+    name = gets.chomp
+    puts 'permission'
+    perm = gets.chomp
+    x = Student.new(cls, age, name, permission: perm)
+    @people.push(x)
+    menu
+  end
+
+  def create_a_teacher
+    puts 'specialization'
+    spec = gets.chomp
+    puts 'age'
+    age = gets.chomp
+    puts 'name'
+    name = gets.chomp
+    x = Teacher.new(spec, age, name)
+    @people.push(x)
+    menu
+  end
+
+  def create_a_book
+    puts 'title'
+    titl = gets.chomp
+    puts 'author'
+    auth = gets.chomp
+    x = Book.new(titl, auth)
+    @book.push(x)
+    menu
+  end
+
+  def create_a_rental
+    puts 'date(yyyy/mm/dd)'
+    date = gets.chomp
+    puts 'select a book'
+    @book.each_with_index { |b, i| puts "#{i}) #{b.title} written by #{b.author}" }
+    bookind = gets.chomp.to_i
+    book = @book[bookind]
+    puts 'select a person'
+    @people.each_with_index { |p, i| puts "#{i}) #{p.name}" }
+    personind = gets.chomp.to_i
+    person = @people[personind]
+    rental = Rental.new(date, book, person)
+    @rental.push(rental)
+    menu
+  end
+
+  def list_all_rentals
+    if @rental.empty?
+      puts("\n\nThere are no rentals\n\n")
+    else
+      puts 'please type person id'
+      id = gets.chomp.to_i
+      @rental.each do |r|
+        puts "\n\n#{r.book} rented by #{r.person} on #{r.date}\n\n" if r.person.id == id
+      end
+    end
```

Comment:
- [ ] The school library console app should behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) provided for you. But the ` list_all_rentals` option does not work as expected, this is the screenshot 👇🏻 The book title and author does not show correctly, Kindly fix this 👍🏻 



- Your console:

![Screenshot from 2022-12-22 18-14-43](https://user-images.githubusercontent.com/79058364/209180897-175c7498-6bf3-4005-a611-75d57d195a6f.png)


- Video provided:

![Screenshot (1)](https://user-images.githubusercontent.com/79058364/209179860-bf2a4a70-580e-4317-b8b3-59f7df1a9bb6.png)